File: JavaScript/kebabize.js
/*
|--------------------------------------------------------------------------
| Kebabize
|--------------------------------------------------------------------------
|
*/

function kebabize(str) {
   res = '';
   for (let i = 0; i < str.length; i++) {
      res += str[i] != str[i].toUpperCase() ? str[i] : str[i].match('^[0-9]') ? '' : `-${str[i].toLowerCase()}`
   }
   return res.split('-').filter(i => i != '').join('-');
}


function kebabize(str) {
   return str.replace(/[0-9]/g, '').replace(/\B[A-Z]/g, (x) => '-' + x).toLowerCase();
}


function kebabize(str) {
   let nonumber = str.replace(/\d/g, '').split(/(?=[A-Z])/g);
   return nonumber.join('-').toLowerCase();
}


function kebabize(str) {
   return str.replace(/([A-Z])/g, "-$1").replace(/[0-9]/g, '').toLowerCase().replace(/^-/, '');
}


console.log(kebabize('myCamelCasedString'), 'my-camel-cased-string');
console.log(kebabize('myCamelHas3Humps'), 'my-camel-has-humps');
console.log(kebabize('Cora'), 'cora');

File: Python/kebabize.py
"""
|--------------------------------------------------------------------------
| Kebabize
|--------------------------------------------------------------------------
|
"""

from re import match, sub
def kebabize(string):
   res = ''
   for i in range(len(string)):
      res += string[i] if string[i] != string[i].upper() else '' if match('^[0-9]', string[i]) else f'-{string[i].lower()}'
   return '-'.join(list(filter(lambda i: i != '' , res.split('-'))))


def kebabize(s):
   return ''.join(c if c.islower() else '-' + c.lower() for c in s if c.isalpha()).strip('-')


def kebabize(string):
   a = ''
   for i in string:
      if i.isupper():
         a += '-' + i.lower()
      if i.islower():
         a += i
   return a.strip('-')


def kebabize(s):
   return sub('\B([A-Z])', r'-\1', sub('\d', '', s)).lower()


print(kebabize('myCamelCasedString'), 'my-camel-cased-string')
print(kebabize('myCamelHas3Humps'), 'my-camel-has-humps')
print(kebabize('SOS'), 's-o-s')
print(kebabize('42'), '')
print(kebabize('CodeWars'), 'code-wars')